bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -53,28 +54,36 @@ impl Default for Counter {
 }
 
 pub trait Connection: core::fmt::Debug {
-    fn poll<Ctx: Context>(&mut self, context: &mut Ctx) -> Poll<Result<()>>;
+    fn poll_negotiate<Ctx: Context>(&mut self, context: &mut Ctx) -> Poll<Result<()>>;
+    fn poll_action<Ctx: Context, F>(&mut self, context: &mut Ctx, action: F) -> Poll<Result<()>>
+    where
+        F: FnOnce(&mut connection::Connection) -> Poll<Result<usize, crate::error::Error>>;
 }
 
 pub trait Context {
     fn receive(&mut self, max_len: Option<usize>) -> Option<Bytes>;
     fn send(&mut self, data: Bytes);
 }
 
+pub enum Mode {
+    Client,
+    Server,
+}
+
 #[derive(Debug)]
-pub struct Pair<Server: Connection, Client: Connection> {
-    pub server: (Server, MemoryContext),
-    pub client: (Client, MemoryContext),
+pub struct Pair<C: Connection> {
+    pub server: (C, MemoryContext),
+    pub client: (C, MemoryContext),
     pub max_iterations: usize,
```

Comment:
Why was this changed?

Reply:
Essentially because of [this match statement](https://github.com/aws/s2n-tls/pull/4114/files/19a1584f275fc239e5dfe2b7fd06a1f52e0ff4fa#diff-a452b39f26afea3da0803229ca873489c2225790c4d9db68139ba7ee81f6bddaR122). I wanted to be able to add a parameter that states which connection is doing the sending/recving action. However I couldn't do that since the Client and Server were different types. But technically, they currently are the same type so this change doesn't mess anything up.